NeuralNetwork/NetworkModels/Neuron.js

```diff
@@ -0,0 +1,80 @@
+const uuidv4 = require('uuid/v4');
+const Network = require('./Network');
+const Synapse = require('./Synapse');
+const Sigmoid = require('./Sigmoid');
+
+/* 
+    Neuron Class Properties: 
+    public Guid Id { get; set; }
+    public List<Synapse> InputSynapses { get; set; }
+    public List<Synapse> OutputSynapses { get; set; }
+    public double Bias { get; set; }
+    public double BiasDelta { get; set; }
+    public double Gradient { get; set; }
+    public double Value { get; set; }
+*/
+
+const sum = (arr, transform = x => x) => arr.reduce((acc, curr) => {
+    return acc + transform(curr);
+}, 0);
+
+
+class Neuron {
+  /* optionally takes an array of Neurons */
+    constructor(inputNeurons = null) {
+        this.Id = uuidv4();
+        this.InputSynapses = [];
+        this.OutputSynapses = [];
+        this.Bias = Network.GetRandom();
+        this.BiasDelta = 0;
+        this.Gradient = 0;
+        this.Value = 0;
+
+        if (inputNeurons != null) {
+            inputNeurons.forEach(inputNeuron => {
+                const synapse = new Synapse(inputNeuron, this);
+                inputNeuron.OutputSynapses.Add(synapse);
+                this.InputSynapses.Add(synapse);   
+            });
+        }
+    }
+
+    //Values & Weights
+    //public virtual double CalculateValue()
+    CalculateValue() {
+        const inputSynapsesValueSum = sum(this.InputSynapses, synapse => synapse.Weight * synapse.InputNeuron.Value);
+        this.Value = Sigmoid.Output(inputSynapsesSum + this.Bias);
```

Comment:
i'm not seeing inputSynapsesSum  anywhere else. should this be inputSynapsesValueSum?

Reply:
yes indeed